### ISSUE TITLE
Issue with streaming with auth

### ISSUE THREAD

**Issue by vitalmon** [pedroSG94/rtmp-rtsp-stream-client-java#835]
Hi @pedroSG94 
I'm using Nimble for the streaming. When I try to do a stream with auth data through the .setAuthorization(login, pass) before starting the stream I receive a 'Time out connection failed' error. After removing the Push login and Push password on the server and removing the call of method .setAuthorization(login, pass) the streaming works fine. I've got I can't use the endpoint like rtmp://pushlogin:pushpass@ip:1935/app/stream
I have the same app for iOS and the publishing stream with auth data works fine over there.
Can we fix it or figure out the cause?
Thanks a lot.

**Comment by marcin-adamczewski** [comment]
@vitalmon You have to call setAuthorization() before starting the stream.

**Comment by vitalmon** [comment]
> @vitalmon You have to call setAuthorization() before starting the stream.

yes. I have

**Comment by vitalmon** [comment]
> @vitalmon You have to call setAuthorization() before starting the stream.

Should I call it after the .prepsreAudio() and .prepareVideo() methods?

**Comment by 1-CharlieShin** [comment]
I am testing via this app for RTSP streaming.
So I added a line in `DisplayService.kt` file.

`
fun init(context: Context) {
      contextApp = context
      if (displayBase == null) {
        displayBase = RtmpDisplay(context, true, connectCheckerRtp)
      }

      displayBase?.setAuthorization("my_id", "my_password")
    }
`
I added `displayBase?.setAuthorization("my_id", "my_password")`.
But I see only "Stream auth error" when I put address in the TextBox.

Does the `setAuthorization` function work in other IP Camera except Wowza server?

Sorry for short english. Thanks.
